### DISCUSSION TITLE
Add a timeout before signaling job cancellation when shutting down

### DISCUSSION THREAD

**Discussion by kmcc049** [quartznet/quartznet#2066] [Ideas]
Currently there is no way that you can configure the shutdown operation to give say ten seconds of time for all jobs to complete before cancelling the token. The options seem to be either 

- Just wait forever
- Cancel them all immediately

I'd like to be able to set some TimeSpan value on IServiceCollectionQuartzConfigurator that could be used in conjunction with InterruptJobsOnShutdownWithWait to wait for that timespan before cancelling the jobs.

**Comment by jafin** [comment]
Will you be competing with the frameworks shutdown timeout? Which if I read is about 5 seconds?  
https://andrewlock.net/extending-the-shutdown-timeout-setting-to-ensure-graceful-ihostedservice-shutdown/

**Comment by kmcc049** [comment]
You can extend that with `services.Configure<HostOptions>(opts => opts.ShutdownTimeout = TimeSpan.FromSeconds(x));` 
Currently quartz's behaviour competes with it anyway where it just waits infinitely for each job if you set WaitForComplete = true, without setting InterruptJobsOnShutdownWithWait = true
